llm_on_ray/inference/predictors/hpu_predictor.py

```diff
@@ -181,6 +184,7 @@ def _process_config(self, config):
 
     def get_streamer(self):
         if self.infer_conf.deepspeed:
+            # Q2: Why always use the first worker?
             return ray.get(self.deepspeed_workers[0].get_streamer.remote())
```

Comment:
Hi @xwu99 , Could you please help explain this question, I think it is the same idea as in deepspeed_predictor.py.

Reply:
This is a distributed inference, worker 0 is assigned as rank 0 (according to the standard MPI model) which is the main rank to return the result, other ranks only engage in calculation, not returning the result. In fact, all ranks holding the same result in this case. You can consider rank 0 is head process of the distributed worker group.